docs/user-guide/scaling.rst

```diff
@@ -0,0 +1,319 @@
+.. _MongoDB: http://www.mongodb.org/
+.. _Apache Qpid: https://qpid.apache.org/
+.. _RabbitMQ: http://www.rabbitmq.com/
+.. _MongoDB Deployment: http://www.mongodb.org/about/introduction/#deployment-architectures
+.. _Apache Qpid HA docs: https://qpid.apache.org/releases/qpid-0.28/cpp-broker/book/chapter-ha.html
+.. _RabbitMQ HA docs: http://www.rabbitmq.com/ha.html
+.. _mod_status: https://httpd.apache.org/docs/2.2/mod/mod_status.html
+.. _HAProxy: http://www.haproxy.org/
+
+Scaling Pulp
+============
+
+Great effort has been put into Pulp to make it scalable. A default Pulp
+install is an "all-in-one" style setup with everything running on one machine.
+However, Pulp supports a clustered configuration with different servers or
+containers to increase availability and performance.
+
+Overview of Pulp components
+---------------------------
+
+Pulp consists of several components:
+
+* ``httpd`` - The webserver process serves published repositories and handles
+  Pulp REST API requests. Simple requests like repository creation are handled
+  immediately whereas longer tasks are asynchronously processed by a worker.
+
+* ``pulp_workers`` - Worker processes handle longer running tasks
+  asynchronously, like repository publishes and syncs.
+
+* ``pulp_celerybeat`` - The celerybeat process discovers and monitors workers.
+  Additionally, it performs task cancellations in the event of a worker
+  shutdown or failure. The celerybeat process also initiates scheduled tasks,
+  and automatically cancels tasks that have failed more than *X* times. This
+  process also initiates periodic jobs that Pulp runs internally. In a Pulp
+  cluster, exactly one of these should be running!
+
+* ``pulp_resource_manager`` - The resource manager assigns tasks to workers,
+  and ensures multiple conflicting tasks on a repo are not executed at the same
+  time. In a Pulp cluster, exactly one of these should be running!
+
+Additionally, Pulp relies on other components:
+
+* `MongoDB`_ - the database for Pulp
+
+* `Apache Qpid`_ or `RabbitMQ`_ - the queuing system that Pulp uses to assign
+  work to workers. Pulp can operate equally well with either Qpid or RabbitMQ.
+
+.. warning:: It is critical to note that ``pulp_celerybeat`` and
+    ``pulp_resource_manager`` should *never* have more than a single instance
+    running under any circumstance!
```

Comment:
I've been looking into this code area recently, and I don't think it would cause corruption in any situation. I hope to study this more formally soon, but I believe corruption will never occur and correct operation should occur as long as all machines running `pulp_celerybeat` have clocks that are within a few minutes of each other. `pulp_resource_manager` should be safe to run in multiple places currently already.